pkg/simulation/simulation.go

```diff
@@ -20,16 +21,18 @@ type SimEnv struct {
 	AgentsPerFloor int
 	logger         log.Entry
 	dayInfo        *day.DayInfo
+	healthInfo     *health.HealthInfo
 	world          world.World
 	custAgents     map[string]agent.Agent
 }
 
-func NewSimEnv(foodOnPlat float64, agentCount []int, agentHP, agentsPerFloor int, dayInfo *day.DayInfo) *SimEnv {
+func NewSimEnv(foodOnPlat float64, agentCount []int, agentHP, agentsPerFloor int, dayInfo *day.DayInfo, healthInfo *health.HealthInfo) *SimEnv {
```

Comment:
Refer to comment on necessity on having this simulation, rather than instantiation in tower directly.
If for input from front end to change these params for different 'difficulties' of simulation, I would argue its better to have three (arbitrary) different levels that the front end can choose between and these are passed into the tower.

Reply:
Comment: Why does simulation have a pointer to healthInfo? Doesn't seem necessary unless I'm missing something - its just used in the tower so can it not be instantiated there?

Reply:
I just followed what was done before for `DayInfo`, as the two structs are quite similar in their use. But I must admit that I am not sure if this pointer is needed for `NewSimEnv`. I think it is actually:
->We create a struct `HealthInfo` in `main.go`, which is then given to `simEnv`. 
->when calling `simEnv.Simulate()`, one of the first step is to create the tower with `t := infra.NewTower(sE.FoodOnPlatform, totalAgents, sE.AgentsPerFloor, sE.dayInfo, sE.healthInfo)` (in `simulation.go)`.
-> we use the struct `HealthInfo` "attached" to the tower in `tower.go`, when calling `hpDecay` (which take the following argument: `func (t *Tower) hpDecay())`, so a point to tower, which in turn contains a `HealthInfo` struct

Reply:
I might be wrong as I did not work on this part of the code. I based my implementation on `DayInfo` regarding its link to tower/simulation as I said.

Reply:
I think this implementation works for parametrisation